src/OrchardCore/OrchardCore.Shortcodes.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Shortcodes;
+
+namespace OrchardCore.Shortcodes
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddShortcode<T>(this IServiceCollection services, string name) where T : class, IShortcodeProvider =>
+            services.AddShortcode(name, null);
+
+        public static IServiceCollection AddShortcode<T>(this IServiceCollection services, string name, Action<ShortcodeOption> describe) where T : class, IShortcodeProvider
+        {
+            services.Configure<ShortcodeOptions>(options => options.AddShortcode(name, describe));
+            services.AddScoped<T>();
```

Comment:
Hi @deanmarcussen ,

This is breaking the feature of image short code. After this, ImageShortcodeProvider is never registered as a  IShortcodeProvider in the DI. This should be changed to 
```csharp
services.AddScope<IShortcodeProvider,T>();

```